docs/source/developers/python.rst

```diff
@@ -344,8 +349,11 @@ Now, build pyarrow:
    python setup.py build_ext --inplace
    popd
 
-If you did not build one of the optional components, set the corresponding
-``PYARROW_WITH_$COMPONENT`` environment variable to 0.
+If you did build one of the optional components (in C++), you need to set the
+corresponding ``PYARROW_WITH_$COMPONENT`` environment variable to 1.
+
+If you wish to delete pyarrow build before rebuilding navigate to the ``arrow/python``
+folder and run ``python setup.py clean``.
```

Comment:
`git clean -Xfd .` is probably more thorough.

Reply:
Adding both possibilities here also?